warp/service.go

```diff
@@ -98,20 +95,21 @@ func (a *API) GetBlockAggregateSignature(ctx context.Context, blockID ids.ID, qu
 }
 
 func (a *API) aggregateSignatures(ctx context.Context, unsignedMessage *warp.UnsignedMessage, quorumNum uint64, subnetIDStr string) (hexutil.Bytes, error) {
-	subnetID := a.sourceSubnetID
+	subnetID := a.chainContext.SubnetID
 	if len(subnetIDStr) > 0 {
 		sid, err := ids.FromString(subnetIDStr)
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse subnetID: %q", subnetIDStr)
 		}
 		subnetID = sid
 	}
-	pChainHeight, err := a.state.GetCurrentHeight(ctx)
+	validatorState := a.chainContext.ValidatorState
+	pChainHeight, err := validatorState.GetCurrentHeight(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	state := warpValidators.NewState(a.state, a.sourceSubnetID, a.sourceChainID, a.requirePrimaryNetworkSigners())
+	state := warpValidators.NewState(validatorState, subnetID, a.chainContext.ChainID, a.requirePrimaryNetworkSigners())
```

Comment:
Was this intentionally changed to be `subnetID` rather than `a.chainContext.SubnetID`?

Reply:
no, good catch 👍